src/native/abbreviateNumber.ts

```diff
@@ -18,9 +14,61 @@ export default new NativeFunction({
             required: true,
             type: ArgType.Number,
         },
+        {
+            name: "decimal places",
+            description: "The number of decimal places to round to",
+            rest: false,
+            type: ArgType.Number,
+        },
     ],
     brackets: true,
-    execute(_, [n]) {
-        return Return.success(Formatter.format(n))
+    execute(_, [n, dp]) {
+        dp ??= 2
+        const map: [number, string][] = [
+                [3, "K"],
+                [6, "M"],
+                [9, "B"],
+                [12, "T"],
+                [15, "Qa"],
+                [18, "Qi"],
+                [21, "Sx"],
+                [24, "Sp"],
+                [27, "Oc"],
+                [30, "No"],
+                [33, "Dc"],
+                [36, "Ud"],
+                [39, "Dd"],
+                [42, "Td"],
+                [45, "Qad"],
+                [48, "Qid"],
+                [51, "Sxd"],
+                [54, "Spd"],
+                [57, "Ocd"],
+                [60, "Nod"],
+                [63, "Vg"],
+                [66, "Uvg"],
+                [69, "Dvg"],
+                [72, "Tvg"],
+                [75, "Qavg"],
+                [78, "Qivg"],
+                [81, "Sxvg"],
+                [84, "Spvg"],
+                [87, "Ocvg"],
+                [90, "Novg"],
+                [93, "Tg"],
+                [96, "Utg"],
+                [99, "Dtg"],
+                [100, " googol"],
+            ],
```

Comment:
place this array outside, otherwise this has to be allocated on the heap every time the function is called